src/sentry/static/sentry/app/components/modals/sudoModal.tsx

```diff
@@ -0,0 +1,170 @@
+import {withRouter} from 'react-router';
+import React from 'react';
+import {ModalHeader, ModalBody} from 'react-bootstrap';
+import {WithRouterProps} from 'react-router/lib/withRouter';
+import styled from '@emotion/styled';
+
+import {t} from 'app/locale';
+import Alert from 'app/components/alert';
+import withApi from 'app/utils/withApi';
+import Button from 'app/components/button';
+import ConfigStore from 'app/stores/configStore';
+import Form from 'app/views/settings/components/forms/form';
+import InputField from 'app/views/settings/components/forms/inputField';
+import {IconFlag} from 'app/icons';
+import TextBlock from 'app/views/settings/components/text/textBlock';
+import U2fContainer from 'app/components/u2f/u2fContainer';
+import space from 'app/styles/space';
+import {Client} from 'app/api';
+
+type Props = WithRouterProps & {
+  api: Client;
+  closeModal: () => void;
+  Header: typeof ModalHeader;
+  Body: typeof ModalBody;
+  /**
+   * User is a superuser without an active su session
+   */
+  superuser?: boolean;
+  /**
+   * expects a function that returns a Promise
+   */
+  retryRequest?: () => Promise<any>;
+};
+
+type State = {
+  error: boolean;
+  busy: boolean;
+};
+class SudoModal extends React.Component<Props, State> {
```

Comment:
nit: Line break before class declaration (maybe we configure https://eslint.org/docs/rules/padding-line-between-statements for it)

Reply:
sometimes I cann't see this line breaks in my code editor, I will have to figure out what is going on. I'll fix it...thanks :)